src/cross_section.F90

```diff
@@ -508,6 +559,189 @@ pure function find_energy_index(mat, E) result(i)
+
+  subroutine multipole_eval_allocate()
+    allocate(sigT_factor(max_L))
+    allocate(twophi(max_L))
+    allocate(broadened_polynomials(max_poly))
+
+    mp_already_alloc = .true.
+  end subroutine
+
+!===============================================================================
+! MULTIPOLE_EVAL evaluates the windowed multipole equations for cross
+! sections in the resolved resonance regions
+!===============================================================================
+
+  subroutine multipole_eval(multipole, Emev, sqrtkT_, sigT, sigA, sigF)
+    type(MultipoleArray), intent(in) :: multipole ! The windowed multipole
+                                                  !  object to process.
+    real(8), intent(in)              :: Emev      ! The energy at which to
+                                                  !  evaluate the cross section
+                                                  !  in MeV
+    real(8), intent(in)              :: sqrtkT_   ! The temperature in the form
+                                                  !  sqrt(kT (in MeV)), at which
+                                                  !  to evaluate the XS.
+    real(8), intent(out)             :: sigT      ! Total cross section
+    real(8), intent(out)             :: sigA      ! Absorption cross section
+    real(8), intent(out)             :: sigF      ! Fission cross section
+    complex(8) :: psi_chi  ! The value of the psi-chi function for the
+                           !  asymptotic form
+    complex(8) :: c_temp   ! complex temporary variable
+    complex(8) :: w_val    ! The faddeeva function evaluated at Z
+    complex(8) :: Z        ! sqrt(atomic weight ratio / kT) * (sqrt(E) - pole)
+    real(8) :: sqrtE       ! sqrt(E), eV
+    real(8) :: invE        ! 1/E, eV
+    real(8) :: dopp        ! sqrt(atomic weight ratio / kT) = 1 / (2 sqrt(xi))
+    real(8) :: temp        ! real temporary value
+    real(8) :: E           ! energy, eV
+    real(8) :: sqrtkT      ! sqrt(kT (in eV))
+    integer :: i_pole      ! index of pole
+    integer :: i_poly      ! index of curvefit
+    integer :: i_window    ! index of window
+    integer :: startw      ! window start pointer (for poles)
+    integer :: endw        ! window end pointer
+
+    ! ==========================================================================
+    ! Bookkeeping
+
+    ! Convert to eV.
+    E = Emev * 1.0e6_8
+    sqrtkT = sqrtkT_ * 1.0e3_8
+
+    ! Define some frequently used variables.
+    sqrtE = sqrt(E)
+    invE = ONE / E
+    dopp = multipole % sqrtAWR / sqrtkT
+
+    if (.not. mp_already_alloc) then
+      call multipole_eval_allocate()
+    end if
+
+    ! Locate us.
+    i_window = floor((sqrtE - sqrt(multipole % start_E)) / multipole % spacing &
+         + ONE)
+    startw = multipole % w_start(i_window)
+    endw = multipole % w_end(i_window)
+
+    ! Fill in factors.
+    if (startw <= endw) then
+      call fill_factors(multipole, sqrtE, sigT_factor, twophi, &
```

Comment:
Looks like `twophi` is never actually used here in `multipole_eval`, so in fact it could be a local variable in `fill_factors`